fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PipeAnalyzer.java

```diff
@@ -164,6 +167,14 @@ public static void analyze(CreatePipeStmt stmt, ConnectContext context) {
         stmt.setInsertSql(insertSql);
         InsertAnalyzer.analyze(insertStmt, context);
 
+        analyzePipeName(stmt.getPipeName(), insertStmt.getTableName().getDb());
+
+        if (!stmt.getPipeName().getDbName().equalsIgnoreCase(insertStmt.getTableName().getDb())) {
+            ErrorReport.reportSemanticException(ErrorCode.ERR_BAD_PIPE_STATEMENT,
```

Comment:
May stmt.getPipeName().getDbName() return null?

Reply:
No. The analyzePipeName will always set the database name of stmt.